src/thor/triplegbuilder.cc

```diff
@@ -481,6 +481,40 @@ void AddTransitNodes(TripLeg_Node* trip_node,
   }
 }
 
+void SetTripEdgeRoadClass(TripLeg_Edge* trip_edge,
+                          const DirectedEdge* directededge,
+                          const GraphTile* graphtile,
+                          GraphReader& graphreader) {
+  trip_edge->set_road_class(GetTripLegRoadClass(directededge->classification()));
+  // If this is a ramp it may have been reclassified in graph enhancer.
+  // To restore the original road class for motorway_links, we check if any of the adjacent edges is
+  // a motorway.
+  if (directededge->use() == Use::kRamp) {
+    auto edge_nodes = graphreader.GetDirectedEdgeNodes(graphtile, directededge);
```

Comment:
maybe we shouldnt use this shortcut but should rather get the opposing edge and then couple the provided tile and the opposing edges end node togther and couple opposing edges tile and the provided edges end node together. the reason i say this is because below, you use graphtile for everything but technically the end node of the provided edge could be in another tile (the tile the opposing edge is in for sure).

Reply:
Pushed up some changes, can you take another look?